pmix/workbook.py

```diff
@@ -81,7 +81,8 @@ def write_out(self, path, strings=False):
             strings (bool): False if the original value should be written,
                 otherwise the string value of the cell is used.
         """
-        wb = xlsxwriter.Workbook(path)
+        filepath = path + 'result.xlsx' if os.path.isdir(path) else path
+        wb = xlsxwriter.Workbook(filepath)
```

Comment:
The `write_out` routine here should not know how to deal with directories.

Reply:
I disagree.

Reply:
It is the wrong level of abstraction. Take a look at `xlsxwriter` itself. https://xlsxwriter.readthedocs.io/workbook.html and ask yourself why they don't allow you to give a directory. Another reason why it isn't a good abstraction is the fact that you are using a "magic value" with `result.xlsx`. Plus, there is currently only one routine that we allow to call this using a directory. That is hardly justification to worry about "every single tool"

Reply:
@jkpr It will be easier to enumerate these points:
i. **Correct level of abstraction**: I agree that `xlsxwriter.Workbook` is the wrong level of abstraction. But I believe that our wrapper class in pmix is a good / correct level of abstraction for this. If not, then we'd need another wrapper class around our wrapper class, and I'm just not sure how useful that would be.
ii. **Magic value**: I think there's no way around that. Any situation I've seen where a file is written out but no file name has been given will have to choose (and does choose) a default value.
iii. **DRYness and UX**: _Borrow_ isn't the only tool that uses `pmix.Workbook`. _Numbering_ and `workbook_cli()` also use it and also accept `--outpath` via the CLI. XLSDiff uses it, though it automatically adds a `-diff` to the file name for now. The fact that we allow _Borrow_ to accept a directory as an `outpath` is arbitrary; why do it for one of our tools and not others? Basically, we want to lower barrier of entry, so we want to allow anyone who might be using our CLI to simply supply a directory and not get an error message (or worse, an uncaught error). This has already happened to both myself and Bciar, so it would surely happen to others (less experienced than us). If that's the case, might as well centralize the logic, rather than implement this for each module that we currently have or may have in the future.

Reply:
@jkpr Going to make your suggested change for the purpose of getting this version out and integrated with our other tools easier. I'll make a separate issue or PR for this.

Reply:
@jkpr nvm, forgot it already exists